app/src/main/java/com/gophillygo/app/activities/AttractionDetailActivity.java

```diff
@@ -88,4 +91,17 @@ public Drawable getFlagImage(AttractionInfo attractionInfo) {
 
         return ContextCompat.getDrawable(this, attractionInfo.getFlagImage());
     }
+
+    public void setupCarousel(CarouselView carouselView, Attraction attraction) {
+        carouselView.setImageListener((position, imageView) -> {
+            String url;
+            if (position == 0) {
+                url = attraction.getWideImage();
+            } else {
+                url = attraction.getExtraWideImages().get(position - 1);
+            }
+            Glide.with(this).load(url).into(imageView);
+        });
+        carouselView.setPageCount(attraction.getExtraWideImages().size() + 1);
```

Comment:
Although this works, consider a less fragile approach than relying on the carousel's page count to avoid NPEs/indexing out of range errors on attempting to access extra images that may not exist by positional offset.

Reply:
How about re-using the `wideImage` if we somehow get a position outside the bounds of `extraWideImages`?

Reply:
Sure, that would be good, with a log error for the unexpected case.